### PR TITLE
Detect Authorization plugin and include it in bake output

### PR DESCRIPTION
**Pull request by markstory**
If the host application has Authorization plugin installed, we can autodetect that and include component checks in generated code from bake.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -16,6 +16,3 @@ indent_size = 2
 
 [phars.xml]
 indent_size = 2
-
-[*.twig]
-insert_final_newline = false
```
Comment (othercorey) [inline .editorconfig]: There was some file that needed this, but I can't remember which one.

File: templates/bake/element/Controller/index.twig [hunk 2]
```diff
@@ -26,6 +26,9 @@
             ->contain({{ Bake.exportArray(belongsTo)|raw }});
 {% else %}
         $query = $this->{{ currentModelName }}->find();
+{% endif %}
+{% if Bake.hasPlugin('Authorization') %}
+        $query = $this->Authorization->applyScope($query);
```
Comment (othercorey) [inline templates/bake/element/Controller/index.twig:31]: Do we need to bake stub code for these?

Comment (markstory) [reply templates/bake/element/Controller/index.twig:31]: I think it is helpful. I usually use this to build in permissions like 'view all the projects I have membership in'.

Comment (othercorey) [reply templates/bake/element/Controller/index.twig:31]: I mean, should we bake/generate the authorization code stubs to match this?

Comment (markstory) [reply templates/bake/element/Controller/index.twig:31]: > I mean, should we bake/generate the authorization code stubs to match this?

That can be done already with `bake policy --type table Articles`. If we wanted to get fancy we could make the bake code generation respect whether or not you have a table/entity policy generated.

File: templates/bake/element/Controller/add.twig [hunk 3]
```diff
@@ -22,6 +22,9 @@
     public function add()
     {
         ${{ singularName }} = $this->{{ currentModelName }}->newEmptyEntity();
+{% if Bake.hasPlugin('Authorization') %}
```
Comment (ADmad) [inline templates/bake/element/Controller/add.twig:25]: I am curious about the use case for authorizing using an empty entity. Personally I have only authorized the "add" action.

Comment (markstory) [reply templates/bake/element/Controller/add.twig:25]: Applications could have roles or rules on who can create what. I agree that most of the time add() will have a policy of `return true`.

File: .editorconfig (hunk 1 above)
Comment (ADmad) [reply .editorconfig]: Yeah whitespace changes in templates can affect the display, so I wouldn't auto add/remove it.

**Comment by markstory** [comment]
Anything else I should take care of with this? I'd like to get this merged so I can move forwards with improving the authentication integration.

File: .editorconfig (hunk 1 above)
Comment (ADmad) [reply .editorconfig]: @markstory Can you revert this please.

**Review by ADmad — APPROVED** [review]
LGTM except the change to `.editorconfig`.

**Pull request merged**